[Doc] Add cross library en doc. (#703)
open-mmlab/mmengine: docs/en/advanced_tutorials/cross_library.md

# Use modules from other libraries

Based on MMEngine's [Registry](registry.md) and [Config](config.md), users can build modules across libraries.
For example, use [MMClassification](https://github.com/open-mmlab/mmclassification)'s backbones in [MMDetection](https://github.com/open-mmlab/mmdetection), or [MMDetection](https://github.com/open-mmlab/mmdetection)'s data transforms in [MMRotate](https://github.com/open-mmlab/mmrotate), or using [MMDetection](https://github.com/open-mmlab/mmdetection)'s detectors in [MMTracking](https://github.com/open-mmlab/mmtracking).

> Modules which registered in the same registry tree can be called across libraries by adding the **package name prefix** before the module's type in the config. Here are some common examples:

Reviewer: ```suggestion
Modules registered in the same registry tree can be called across libraries by adding the **package name prefix** before the module's type in the config. Here are some common examples:
```

      drop_path_rate=0.4,
      layer_scale_init_value=1.0,
      gap_before_final_norm=False,
      init_cfg=dict(
          type='Pretrained',
          checkpoint=
          'https://download.openmmlab.com/mmclassification/v0/convnext/downstream/convnext-tiny_3rdparty_32xb128-noema_in1k_20220301-795e9634.pth',
          prefix='backbone.')),
  neck=dict(...),
  rpn_head=dict(...))
```

## Use data transform across libraries:

As with the example of backbone above, cross-library calls can be simply achieved by adding custom_imports and prefix in the config:

```python
# Use custom_imports to register mmdet transforms to the registry
custom_imports = dict(imports=['mmdet.datasets.transforms'], allow_failed_imports=False)

# Add mmdet prefix to enable cross-library mechanism
train_pipeline=[
    dict(type='mmdet.LoadImageFromFile'),
    dict(type='mmdet.LoadAnnotations', with_bbox=True, box_type='qbox'),
    dict(type='ConvertBoxType', box_type_mapping=dict(gt_bboxes='rbox')),
    dict(type='mmdet.Resize', scale=(1024, 2014), keep_ratio=True),
    dict(type='mmdet.RandomFlip', prob=0.5),
    dict(type='mmdet.PackDetInputs')
]
```

## Use detector across libraries:

> Using an algorithm across libraries is a little bit complex.

Reviewer: ```suggestion
Using a module from another library is a little bit complex.
```
Maybe 'module' here is more proper than 'algorithm'?

Author: This example is about using a detection algorithm whose complexity differs from a single module.

Reviewer: Makes sense to me
